changelog.d/9868.bugfix

```diff
@@ -0,0 +1 @@
+Fix a long-standing bug where errors from federation did not propagate to the client.
```

Comment:
The changes all look sensible, but I'm not entirely sure which errors which were previously swallowed are now propagated correctly. Do you have any examples?

Reply:
Well the rationale for the work is the 403 error in #9814 is now passed back to the client as a 403, instead of a 400.

Reply:
I should probably mention that this is tested by matrix-org/complement#99.